docs/source/API/core/MultiGPUSupport.rst

```diff
@@ -0,0 +1,117 @@
+allocating device specific data.
+
+Basic Usage
+-----------
+
+For allocating and launching kernels on a chosen device, the user is responsible for creating device specific streams to pass to backend constructor.
+Once these are created, they can be passed as any other execution space instance to create policies, allocated managed views, asynchronous deep copies, etc. For creating unmanaged
+views, the user must manually allocate device specific memory to pass to the view constructor.
+
+Contructing Execution Space
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cuda
+^^^^
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    cudaGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    cudaStream_t stream;
+    cudaSetDevice(N);
+    cudaStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::Cuda exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    cudaSetDevice(N);
+    cudaStreamDestroy(stream);
+
+HIP
+^^^
+
+.. warning:: Multi-GPU only supported for rocm 5.6 and later. Because of the lack of HIP API functions for querying a stream's device
+             in pre rocm 5.6, constructing a ``Kokkos::HIP`` instance on the non-default device will not be able to throw an error,
+             problems will just likely arise downstream.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    hipGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    hipStream_t stream;
+    hipSetDevice(N);
+    hipStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::HIP exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    hipSetDevice(N);
+    hipStreamDestroy(stream);
+
+SYCL
+^^^^
+
+.. code-block:: cpp
+
+    // Get list of devices available
+    std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+
+    // Choose some 0 <= N < gpu_devices.size()
+    int N = ...;
+
+    // Create a queue on device N.
+    // Note: Kokkos requires SYCL queues to be "in_order"
+    sycl::queue queue{gpu_devices[N], sycl::property::queue::in_order()};
+
+    // Create execution space
+    Kokkos::SYCL exec_space(queue);
+
+    // Use execution space
+    /* ... */
+
+Notes
+-----
+
+- For ``parallel_reduce(..., const ReducerArgument&... reducer)``, it is important to pass rank-0 views for the ``reducer`` arguments
+  instead of scalar types. Scalar reductions imply a fence on all execution spaces.
```

Comment:
I don't think we promise a global fence but only that the results is available after the call. In any case, this doesn't seem to be specific for multi-GPU setups so I would just remove.